pkg/controller/controller.go

```diff
@@ -169,6 +179,99 @@ func (c *Controller) getWorkflowUpdatesHandler() cache.ResourceEventHandler {
 	}
 }
 
+// This object is responsible for emitting metrics that show the current number of Flyte workflows, cut by project and domain.
+// It contains
+// It needs to be kicked off. The periodicity is not currently configurable because it seems unnecessary.
+type ResourceLevelMonitor struct {
+	Scope promutils.Scope
+
+	// Meta timer - this times each collection cycle
+	CollectorTimer promutils.StopWatch
+
+	// This is a labeled gauge
+	levels *prometheus.GaugeVec
+
+	// The thing that we want to current levels of
+	lister lister.FlyteWorkflowLister
+}
+
+func (r *ResourceLevelMonitor) countList(ctx context.Context, workflows []*v1alpha1.FlyteWorkflow) map[string]map[string]int {
+	// Map of Projects to Domains to counts
+	counts := map[string]map[string]int{}
+
+	// Collect all workflow metrics
+	for i, wf := range workflows {
+		execID := wf.GetExecutionID()
+		if execID.WorkflowExecutionIdentifier == nil {
+			logger.Warningf(ctx, "Workflow does not have an execution identifier! [%v]", wf)
+			continue
+		}
+		fmt.Printf("in the loop %d: %s %s %s %s\n", i, wf.ExecutionID.Project, wf.ExecutionID.Domain, wf.GetID(), wf.ExecutionID.Name)
+		project := wf.ExecutionID.Project
+		domain := wf.ExecutionID.Domain
+		if _, ok := counts[project]; !ok {
+			counts[project] = map[string]int{}
+		}
+		counts[project][domain] += 1
+	}
+
+	return counts
+}
+
+func (r *ResourceLevelMonitor) collect(ctx context.Context) {
+	// Emit gauges at both the project/domain level - aggregation to be handled by Prometheus
+	workflows, err := r.lister.List(labels.Everything())
+	if err != nil {
+		logger.Errorf(ctx, "Error listing workflows when attempting to collect data for gauges %s", err)
+	}
+
+	counts := r.countList(ctx, workflows)
+
+	// Emit labeled metrics, for each project/domain combination. This can be aggregated later with Prometheus queries.
+	metricKeys := []contextutils.Key{contextutils.ProjectKey, contextutils.DomainKey}
+	for project, val := range counts {
+		for domain, num := range val {
+			tempContext := contextutils.WithProjectDomain(ctx, project, domain)
+			gauge, err := r.levels.GetMetricWith(contextutils.Values(tempContext, metricKeys...))
```

Comment:
I presume you verified/test this works? and won't just keep a single value (the latest) around?

Reply:
```
# HELP flyte:propeller:all:collector:collection_cycle_ms Measures how long it takes to run a collection
# TYPE flyte:propeller:all:collector:collection_cycle_ms summary
flyte:propeller:all:collector:collection_cycle_ms{quantile="0.5"} 0
flyte:propeller:all:collector:collection_cycle_ms{quantile="0.9"} 0
flyte:propeller:all:collector:collection_cycle_ms{quantile="0.99"} 0
flyte:propeller:all:collector:collection_cycle_ms_sum 0
flyte:propeller:all:collector:collection_cycle_ms_count 9
# HELP flyte:propeller:all:collector:flyteworkflow Current FlyteWorkflow levels
# TYPE flyte:propeller:all:collector:flyteworkflow gauge
flyte:propeller:all:collector:flyteworkflow{domain="development",project="flyteexamples"} 1
flyte:propeller:all:collector:flyteworkflow{domain="staging",project="flyteexamples"} 1
```